### PR TITLE
Upgrade ScanCode to 30.1.0

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Upgrade the default ScanCode version to 30.1.0, see the commit messages for details.

### CODE REVIEW

File: scanner/src/test/kotlin/scanners/scancode/ScanCodeResultParserTest.kt
Comment (sschuberth) [inline scanner/src/test/kotlin/scanners/scancode/ScanCodeResultParserTest.kt:76]: This (and other test names) should probably consistently refer to either "30.1.0" or "30.*". Or maybe yet better, we should stop referring to the ScanCode version at all, and instead refer to the "output_format_version" written in the JSON, which current is at 1.0.0.

Comment (sschuberth) [reply scanner/src/test/kotlin/scanners/scancode/ScanCodeResultParserTest.kt]: > the old test assets don't have an "output_format_version" so I have kept them as is.

That's because the "output_format_version" JSON property was just recently introduced, see https://github.com/nexB/scancode-toolkit/issues/2653#issuecomment-924934429.

File: scanner/src/test/kotlin/scanners/scancode/ScanCodeResultParserTest.kt
Comment (sschuberth) [inline scanner/src/test/kotlin/scanners/scancode/ScanCodeResultParserTest.kt:275]: There's again this unrelated formatting change.

File: scanner/src/test/kotlin/scanners/scancode/ScanCodeResultParserTest.kt
Comment (sschuberth) [inline scanner/src/test/kotlin/scanners/scancode/ScanCodeResultParserTest.kt:298]: There's again this unrelated formatting change.

File: scanner/src/test/kotlin/scanners/scancode/ScanCodeResultParserTest.kt
Comment (sschuberth) [inline scanner/src/test/kotlin/scanners/scancode/ScanCodeResultParserTest.kt:77]: Missing space before `{`.

File: scanner/src/test/kotlin/scanners/scancode/ScanCodeResultParserTest.kt
Comment (mnonnenmacher) [reply scanner/src/test/kotlin/scanners/scancode/ScanCodeResultParserTest.kt:275]: This time it is intended, see the first commit.

File: scanner/src/test/kotlin/scanners/scancode/ScanCodeResultParserTest.kt
Comment (mnonnenmacher) [reply scanner/src/test/kotlin/scanners/scancode/ScanCodeResultParserTest.kt:298]: This time it is intended, see the first commit.

**Review by sschuberth — APPROVED** [review]

**Comment by sschuberth** [comment]
@tsteenbe please remember to also approve once you're fine with merging.

**Review by tsteenbe — APPROVED** [review]

**Pull request merged**
